postgres/datadog_checks/postgres/statements.py

```diff
@@ -164,6 +164,18 @@ def collect_per_statement_metrics(self, db, db_version, tags):
             self._log.exception('Unable to collect statement metrics due to an error')
             return []
 
+    def _load_pg_stat_statements_safe(self, db, tags):
+        try:
+            self._load_pg_stat_statements(db)
+        except psycopg2.errors.ObjectNotInPrerequisiteState as e:
+            if 'pg_stat_statements must be loaded via shared_preload_librarie' in str(e):
```

Comment:
* Is `librarie` a typo? It will work, but I'm just wondering if there is any reason behind it.
* `e.pgerror` is probably better as the function of the attribute is [well-defined as the actual server error](https://www.psycopg.org/docs/module.html#psycopg2.Error.pgerror) whereas the string representation is arbitrary and could change across versions.

Reply:
Also just to confirm, this is the error message in all PG versions?

Reply:
Before we get into those questions/details, what do you think of @djova 's suggestion [here](https://github.com/DataDog/integrations-core/pull/9628#discussion_r661777393). This would replace this more specific error by something more generic (and maybe a little less immediately useful) but would cover more cases. 
For example, this case would result in a metric like:
```
E     -  MetricStub(name=u'dd.postgres.statement_metrics.error', type=2, value=1.0, tags=['foo:bar', 'server:localhost', 'port:5432', "error:database-<class 'psycopg2.errors.ObjectNotInPrerequisiteState'>-code-55000"], hostname=u'', device=None)]
```
Code:
```
except psycopg2.Error as e:
            self._check.count(
                "dd.postgres.statement_metrics.error", 1, tags=tags + ["error:database-{}-code-{}".format(type(e), e.pgcode)]
            )
            self._log.warning(
                "Unable to collect statement metrics because of an error running queries: %s", e
            )
```

instead of something specific about the actual reason. Note that we can include the `pgcode` (that `code-{}`) or the error type (that example above includes both the error type as well as the postgres code) but each code can have a number of sub reasons.

Reply:
Discussed in Slack. A generic approach is preferred, but we should have some specific codes and pg_stat_statements is a specific code we will catch and indicate.

Reply:
Here's what the metric would look like for the `pg_stat_statements_not_enabled` case:

```
dd.postgres.statement_metrics.error{error:pg_stat_statements_not_enabled}
```

And the generic case (which is just one example using the error _name_):
```
dd.postgres.statement_metrics.error{error:database-ObjectNotInPrerequisiteState}
```